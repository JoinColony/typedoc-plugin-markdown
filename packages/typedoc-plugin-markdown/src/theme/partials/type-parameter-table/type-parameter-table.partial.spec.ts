import { DeclarationReflection, ProjectReflection } from 'typedoc';
import { typeParameterTablePartial } from './type-parameter-table.partial';

describe(`typeParameterTablePartial`, () => {
  let project: ProjectReflection;

  beforeAll(async () => {
    project = global.getProject('generics.ts');
  });

  test(`should compile type parameter table`, () => {
    expect(
      typeParameterTablePartial(
        global.getMockContext(),
        (
          project.findReflectionByName(
            'functionWithTypeParams',
          ) as DeclarationReflection
        )?.signatures[0].typeParameters,
      ),
    ).toMatchSnapshot();
  });
});
