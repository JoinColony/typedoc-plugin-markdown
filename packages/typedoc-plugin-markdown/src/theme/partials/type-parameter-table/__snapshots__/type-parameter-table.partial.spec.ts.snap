// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`typeParameterTablePartial should compile type parameter table 1`] = `
"
| Name | Type | Description |
| :------ | :------ | :------ |
| \`A\` | extends [\`ClassWithTypeParams\`]( ClassWithTypeParams )<\`string\`, \`number\`, [\`A\`]( A )\\\\> | Comment for type \`A\` |
| \`B\` | \`string\` \\\\| \`boolean\` | Comment for type \`B\` |
| \`C\` | \`string\` | - |
"
`;
